Automattic/jetpack
# WPcom: adjust filter for comment counts used in Calypso

Currently certain comment types, `comment`, `pingback`, and `trackback` are whitelisted for being displayed in the comments list. Comment counts, on the other hand are calculated by excluding the comment types `order_note`, `webhook_delivery`, `review`, `action_log`.

This is done using the `jetpack_api_exclude_comment_types_count` filter, and does not take into account other comment types that may be used by third-party plugins. As a result, there can be a discrepancy between the comments listed, and the comment count that are displayed.

After discussing this further with @samiff we decided that it may be best to deprecate `jetpack_api_exclude_comment_types_count`, and instead introduce `jetpack_api_include_comment_types_count`, which will bring parity to the comment types displayed, and the comment types counted, defaulting to including `comment`, `pingback`, and `trackback` comment types.




Fixes [#63872](https://github.com/Automattic/wp-calypso/issues/63872)

#### Changes proposed in this Pull Request:

* deprecate `jetpack_api_exclude_comment_types_count` and introduce `jetpack_api_include_comment_types_count`
* The database is now queried for comment_types `WHERE comment_type IN` matching those specified in the array



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Create several comments on a post/page
* Go to the `wp_comments` table, and edit the `comment_type` to something other than `comment`, `pingback`, `trackback`
* Go to [https://wordpress.com/comments/{domain}](https://wordpress.com/comments/%7Bdomain%7D)
* The comment counts should match the number of comments actually listed
